ert/data/_transformation.py

```diff
@@ -0,0 +1,66 @@
+import pathlib
+from pathlib import Path
+import tarfile, os, io
+import aiofiles
+from ert.serialization import get_serializer
+import stat
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from ert.data import RecordTransmitter
+
+_BIN_FOLDER = "bin"
+
+
+class FileTransformation:
```

Comment:
Seems like a good candidate for an ABC. Unless we really want `dump` to be shared behaviour

Reply:
If not an ABC, I think `pre_dump` and `post_dump` should raise `NotImplementedError`.